src/protocol/boolean/random_bits_generator.rs

```diff
@@ -98,6 +160,18 @@ mod tests {
         let _: Fp31 = result.reconstruct(); // reconstruct() will validate the value.
     }
 
+    #[tokio::test]
+    pub async fn uses_fallback_channel() {
```

Comment:
Not 100% on this test.  This won't be closing any of the channels.

Reply:
yea I only wanted to test that uses that channel - without extra plumbing it is not easy to see which channel is being used. I also think we can test main channel closure easily by setting up a limit on number of records and then exhausting it. Not sure how we would test fallback channel closure though if it happens inside - I could expose `is_closed(&self)` method on the gateway to make it work